src/HydraAuctionOnchain/Types/AuctionTerms.hs

```diff
@@ -48,17 +58,78 @@ newtype PAuctionTerms (s :: S)
 instance DerivePlutusType PAuctionTerms where
   type DPTStrat _ = PlutusTypeData
 
+instance PTryFrom PData PAuctionTerms
+
 data AuctionTerms
 
 type instance PlyArgOf PAuctionTerms = AuctionTerms
 
+pvalidateAuctionTerms :: Term s (PAuctionTerms :--> PUnit)
+pvalidateAuctionTerms = phoistAcyclic $
+  plam $ \auctionTerms -> P.do
+    rec <- pletAll auctionTerms
+
+    -- Auction lot must not include any ADA.
+    passert $(errCode AuctionTerms'Error'AuctionLotNonZeroAda) $
+      padaOf # rec.auctionLot #== 0
+
+    -- All amounts in the auction lot must be positive.
+    passert $(errCode AuctionTerms'Error'NonPositiveAuctionLotValue) $
+      pvaluePositive # rec.auctionLot
+
+    -- The payment part of the seller address must be seller pkh.
+    passert $(errCode AuctionTerms'Error'SellerAddressLacksPubKeyCredential) $
+      pisJust #$ paddrPaymentKeyHash # rec.sellerAddress
+
+    -- TODO: The seller pubkey hash corresponds to the seller verification key.
+    -- Note: this check only becomes possible on-chain in Plutus V3.
+    -- https://github.com/input-output-hk/plutus/pull/5431
```

Comment:
As we moved to addresses, will we still need it?

Reply:
Yes, we should check the correspondence between the two, unless we decide to distinguish between SellerVK (auction gatekeeper) and SellerPKH (beneficiary + oracle) and treat them as different actors.